typescript/lib/services/service_context.ts

```diff
@@ -65,42 +66,48 @@ class ServiceContext {
     }
 
     // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
-    public execCommand(command: string[]): [number, Uint8Array | string, Error] {
+    public async execCommand(command: string[]): Promise<[number, Uint8Array | string, Error]> {
         const serviceId: ServiceID = this.serviceId;
         const args: ExecCommandArgs = newGetExecCommandArgs(serviceId, command);
 
-        var resp: ExecCommandResponse;
-        var error: grpc.ServiceError;
-        this.client.execCommand(args, function(error, feature){ //TODO (comment) - don't use return value, but use callback parameters
-            error = error; 
-            resp = feature;
+        const promiseAsync: Promise<ResultAsync<ExecCommandResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.execCommand(args, (_unusedError: grpc.ServiceError, response: ExecCommandResponse) => {
```

Comment:
Ah, you'll still need to check the `execCommand`'s error, so that you know whether to return `ok()` or `error()`. Otherwise, if the `execCommand` fails (e.g. it can't reach the API container) then we'll never notify the user